### PR TITLE
 feat(debugfiles): Allow to search by code_id or debug_id

### PR DESCRIPTION
**Pull request by jan-auer**
Adds new query options to the debug files endpoint. The Symbolicator will use this endpoint to query and download debug files from Sentry.

### Query Rules

- A debug file is only considered, if it is associated to the project of the request. The symbolicator cannot query across multiple projects to avoid unintentionally mixing symbols from multiple organizations.
- If a code id is provided and a DIF specifies a code id, then the file is included if and only if the code id matches.
- If no code id is provided or a DIF does not specify a code id, then the file is included if and only if the debug id matches.
- The request must either provide a code id or debug id to search for. This is an invariant upheld by the Symbolicator. In case the symbolicator ever launches an invalid request without these ids, the result list is limited to 20 items.
- If multiple files match the above criteria, all of them are returned. Realistically, there cannot be more than 20 results, so paging does not have to be considered.

Based on #12448

### CODE REVIEW

**Comment by jan-auer** [comment]
This PR will be **rebased** once #12448 is merged.

**Review by untitaker — APPROVED** [review]

**Comment by jan-auer** [comment]
Blocked on #12483, in case we decide to provide a dedicated endpoint for the symbolicator.

**Pull request merged**
